core/src/main/resources/error/error-classes.json

```diff
@@ -932,7 +932,7 @@
   },
   "NUM_COLUMNS_MISMATCH" : {
     "message" : [
-      "<operator> can only be performed on tables with the same number of columns, but the first table has <refNumColumns> columns and the <invalidOrdinalNum> table has <invalidNumColumns> columns."
+      "<operator> expects matching number of columns. But the left side (target) has <leftNumCols> while the right side (source) has <rightNumCols>."
```

Comment:
expects matching number of columns -> expect matching numbers of columns

Reply:
cc @srielau @srowen Please, review this error message.

Reply:
I'm thinking of the operator being UNION (hence left and right).
So:
SET
INSERT
UNION
INTERSECT
EXCEPT 
IN (?) -- this may go through struct logic not sure)

Reply:
I would avoid words like `left`/`right` and `reference`/`target`, and just slightly improve the existing error template by replacing the word `table` to let's say `input`:
```
"<operator> can only be performed on inputs with the same number of columns, but the first input has <firstNumColumns> columns and the <invalidOrdinalNum> input has <invalidNumColumns> columns."
```

Reply:
@panbingkun Could you change the error message, please.